txnkv/transaction/2pc.go

```diff
@@ -1019,7 +1019,13 @@ func (c *twoPhaseCommitter) doActionOnGroupMutations(bo *retry.Backoffer, action
 			return nil
 		}
 		c.store.WaitGroup().Add(1)
+		if c.txn.secondaryLockCleanupLifecycleHooks.Pre != nil {
+			c.txn.secondaryLockCleanupLifecycleHooks.Pre()
+		}
 		err = c.store.Go(func() {
+			if c.txn.secondaryLockCleanupLifecycleHooks.Post != nil {
```

Comment:
To avoid hook at all the goroutines, another approach is to hook at the beginning of `execution` function in `2pc.go` and the defer block of it, using `c.store.Waitgroup().wait()` to ensure the background goroutines are finished in the defer hook.

In the current implementation, the `cleanup` goroutine needs to be hooked too.

Reply:
Done. I've added a new `spawn` method for the `txn` as a helper function to track all goroutines. It'll maintain the lifecycle hooks and the inc/dec of  `txn.store.WaitGroup()`.

Also provide a `spawnWithStorePool` to call `txn.store.Go` internally. I don't modify the existing logic about whether to use the store pool.

Reply:
(It's called `spawn` but not `go`, because I want it to be a private function, but `go` is a keyword so I cannot use it as a function name.)